### PR TITLE
374 make predictor_values an argument in GLMPrediction

### LINKED ISSUE #374: Make predictor values an argument passed to `GLMPrediction.predict()` and `GLMPrediction.sample()`
Change `fixed_predictor_values` from `GLMPrediction` class attribute to a `predictor_values` argument. This will allow it to interact more appropriately with numpyro tools such as `Predictive`.

@AFg6K7h4fhy2, @sbidari is going to do this, so when it's ready you can use the fix in your cfaepim branch.

### PR DESCRIPTION
**Pull request by sbidari**
closes #374

### CODE REVIEW

File: model/src/pyrenew/regression.py [hunk 1]
```diff
@@ -114,21 +107,31 @@ def predict(
...
             Array of transformed predictions.
         """
-        transformed_prediction = (
-            intercept + self.fixed_predictor_values @ coefficients
-        )
+        transformed_prediction = intercept + predictor_values @ coefficients
         return self.transform.inv(transformed_prediction)
 
-    def sample(self) -> dict:
+    def sample(self, **kwargs) -> dict:
```
[the remark this answers is not in the document]
Comment (dylanhmorris) [reply model/src/pyrenew/regression.py:126]: I'd explicitly declare `predictor_values` and make it a mandatory positional argument

File: model/src/pyrenew/regression.py [hunk 2]
```diff
@@ -114,21 +107,31 @@ def predict(
...
         """
         Sample generalized linear model
 
+        Parameters
+        -----------
+        **kwargs : dict
```
Comment (dylanhmorris) [inline model/src/pyrenew/regression.py:132]:
```suggestion
        predictor_values : ArrayLike(n_predictors, n_observations)
            Matrix of predictor variables (covariates) for the
            regression problem. Each row should represent the
            predictor values corresponding to an observation;
            each column should represent a predictor variable.
            Do not include values of 1 for the intercept;
            these will be added automatically. Passed as the 
            `predictor_values` argument to 
            :meth:`GLMPrediction.predict()`

        **kwargs : dict
             Additional keyword arguments. Ignored.
```

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
This looks very good, thanks @sbidari. Just one small mandatory change.

Also, if you have the time, it could be in scope to fix the class docstring (which incorrectly refers to "`GLMObservation`" and "`observations`" (legacy of an older version) and/or to change the return type of `sample` from `dict` to `SampledValue`, to be consistent with other `RandomVariable`s. Either/both of these could also be separate PRs.

File: model/src/pyrenew/regression.py [hunk 3]
```diff
@@ -141,7 +155,9 @@ def sample(self) -> dict:
         coefficients = numpyro.sample(
             self.name + self.coefficient_suffix, self.coefficient_priors
         )
-        prediction = self.predict(intercept, coefficients)
+
+        predictor_values = kwargs.get("predictor_values")
```
Comment (dylanhmorris) [inline model/src/pyrenew/regression.py:159]:
```suggestion
```

**Review by dylanhmorris — CHANGES_REQUESTED** [review]

**Comment by sbidari** [comment]
@dylanhmorris ready for re-review!
The recent commit includes mandatory changes plus docstring fix and change to the `sample` return object from a `dict` to a collection of `SampledValues`

File: model/src/pyrenew/regression.py [hunk 4]
```diff
@@ -114,38 +138,54 @@ def predict(
...
             Array of transformed predictions.
         """
-        transformed_prediction = (
-            intercept + self.fixed_predictor_values @ coefficients
-        )
+        transformed_prediction = intercept + predictor_values @ coefficients
         return self.transform.inv(transformed_prediction)
 
-    def sample(self) -> dict:
+    def sample(self, predictor_values: ArrayLike) -> dict:
```
Comment (dylanhmorris) [inline model/src/pyrenew/regression.py:157]:
```suggestion
    def sample(self, predictor_values: ArrayLike) -> GLMPredictionSample:
```

File: model/src/pyrenew/regression.py [hunk 5]
```diff
@@ -114,38 +138,54 @@ def predict(
...
-            coefficients=coefficients,
+        prediction = self.predict(intercept, coefficients, predictor_values)
+
+        return GLMPredictionSample(
+            prediction=SampledValue(prediction),
+            intercept=SampledValue(intercept),
+            coefficients=SampledValue(coefficients),
         )
 
     def __call__(self, **kwargs):
```
Comment (dylanhmorris) [inline model/src/pyrenew/regression.py:191]:
```suggestion
    def __call__(self, *args, **kwargs):
```

Comment (dylanhmorris) [reply model/src/pyrenew/regression.py:191]: And similarly for the call to `sample()` within

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
Looks very good. A few very minor tweaks. Thanks, @sbidari!

File: model/src/pyrenew/regression.py (hunk 4 above)
Comment (sbidari) [reply model/src/pyrenew/regression.py]: changed in cf3b96a7f790b891ec1df0141542bcb9863dd365

File: model/src/pyrenew/regression.py (hunk 5 above)
Comment (sbidari) [reply model/src/pyrenew/regression.py]: included in cf3b96a7f790b891ec1df0141542bcb9863dd365

**Pull request merged**